android/app/src/main/kotlin/com/jabook/app/jabook/audio/data/repository/ChapterMetadataRepository.kt

```
@@ -83,7 +83,19 @@ public class ChapterMetadataRepository
          */
         public suspend fun saveChapters(chapters: List<ChapterMetadataEntity>): Result<Unit> =
             try {
-                chapterDao.upsertChapters(chapters)
+                if (chapters.isEmpty()) {
+                    return Result.Success(Unit)
+                }
+
+                val bookIds = chapters.map { it.bookId }.distinct()
+                require(bookIds.size == 1) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)